kano_settings/system/display.py

```diff
@@ -379,25 +440,30 @@ def read_hdmi_mode():
 
 
 def find_matching_mode(modes, group, mode):
-    string = '{}:{}'.format(group.lower(), mode)
-    for i, m in enumerate(modes):
-        if m.startswith(string):
-            return i + 1
+    """ Get the index of the mode in the list from the HDMI group and HDMI mode. """
+
+    for index, current_mode in enumerate(modes):
+        if current_mode['mode'] == mode and current_mode['group'] == group:
+            return index + 1
 
     # 0 for auto
     return 0
 
 
 def read_edid():
-    edid_dat_path = '/tmp/edid.dat'
+    delete_file(TMP_EDID_DAT_PATH)
```

Comment:
Would we also like a `use_cached` option here?

Reply:
It would probably benefit from it since we don't use HDMI hotplugging currently, but for the purpose of this PR, I didn't want to modify (too much) what we currently had. I think we should do it in another PR.